detect-nuget-inspector/detect-nuget-inspector/Inspection/Util/SolutionDirectoryPackagesPropertyLoader.cs

```diff
@@ -0,0 +1,201 @@
+                    {
+                        packages.Add(new PackageId(packageName, packageVersion));
+                    }
+                }
+            }
+
+            return packages;
+        }
+        
+        public bool GetVersionOverrideEnabled()
+        {
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+
+            XmlDocument doc = new XmlDocument();
+            doc.Load(PropertyPath);
+            
+            XmlNodeList centralVersionOverride = doc.GetElementsByTagName("CentralPackageVersionOverrideEnabled");
+
+            bool checkVersionOverride = !(centralVersionOverride != null && centralVersionOverride.Count > 0 && centralVersionOverride.Item(0).InnerXml == "false");
+
+            return checkVersionOverride;
+        }
+
+        public HashSet<PackageId> GetGlobalPackageReferences()
+        {
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+
+            XmlDocument doc = new XmlDocument();
+            doc.Load(PropertyPath);
+
+            HashSet<PackageId> globalPackageReferences = new HashSet<PackageId>();
+            
+            XmlNodeList propertyGroupTags = doc.GetElementsByTagName("PropertyGroup");
+            
+            Dictionary<string, string> propertyGroups = GetPropertyGroups(propertyGroupTags);
+
+            bool managePackageVersionsCentrally = propertyGroups.ContainsKey("ManagePackageVersionsCentrally")
+                ? Boolean.Parse(propertyGroups["ManagePackageVersionsCentrally"])
+                : true;
+            
+            XmlNodeList globalPackageReferenceNodes = doc.GetElementsByTagName("GlobalPackageReference");
+
+            if (managePackageVersionsCentrally && globalPackageReferenceNodes != null && globalPackageReferenceNodes.Count > 0)
+            {
+               globalPackageReferences = GetPackageVersions(globalPackageReferenceNodes,propertyGroups);
+            }
+
+            return globalPackageReferences;
+        }
+
+        public string GetPropsFilePath(XmlNodeList importNodes)
+        {
+            if (importNodes != null && importNodes.Count > 0)
+            {
+                foreach (XmlNode node in importNodes)
+                {
+                    XmlAttributeCollection attributes = node.Attributes;
+
+                    foreach (XmlAttribute attr in attributes)
+                    {
+                        if (attr.LocalName.Contains("Project"))
+                        {
+                            return attr.Value;
+                        }
+                    }
+                }
+            }
+            return null;
+        }
+
+        public Dictionary<string, string> GetPropertyGroups(XmlNodeList propertyGroups)
+        {
+            Dictionary<string, string> groups = new Dictionary<string, string>();
+
+            if (propertyGroups != null && propertyGroups.Count > 0)
```

Comment:
This check is quite common. Could use a helper util if a native method does not exist to check if a list is empty.